### PR TITLE
Birthchart calculation

### PR DESCRIPTION
**Pull request by jfschwarz**
The PR adds the calculation of the birthchart to the playground app to show how it can be integrated into the future minting app

### CODE REVIEW

**Comment by samepant** [comment]
this is looking awesome!! 

one thing i've noticed:
- aspect lines that are within ~5 points of being perfectly horizontal or vertical mess up the gradients applied to the aspect lines -__- 
Screen Shot 2022-05-17 at 10 48 33 AM
Screen Shot 2022-05-17 at 10 46 27 AM
Is there an easy way to enforce not straight lines (x1 - x2 > 5) (y1 - y2 > 5) ?

File: app/birthchart/planets.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  latitude: number,
+  longitude: number,
+  altitude = 0,
+  date = new Date()
+) => {
+  const positions = calculateBodyPositions(latitude, longitude, altitude, date);
+  const planetPositions = Object.values(positions).slice(2); // skip sun and moon
+  return planetPositions.map(({ x, y }) => {
+    // coordinates further away than the scaled radius mean that the planet is not currently visible on the local celestial sphere
+    // we zero them, which means they won't be rendered
```
Comment (samepant) [inline app/birthchart/planets.ts:58]: this Zeroing of coordinates is resulting in the planet being at the center of the birthchart
Screen Shot 2022-05-18 at 12 06 13 PM

File: app/components/settings/LocationInput.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  latitude: number;
+  longitude: number;
+}
+
+interface Props {
+  value: Location | undefined;
+  onChange(value: Location | undefined): void;
+}
+
+const defaultLocation = fetch("http://ip-api.com/json")
```
Comment (samepant) [inline app/components/settings/LocationInput.tsx:14]: this works locally, but chrome doesn't like cross origin non-ssl requests, is there a way to get it through the browser?

**Review by samepant — CHANGES_REQUESTED** [review]

File: app/components/settings/LocationInput.tsx (hunk 2 above)
Comment (jfschwarz) [reply app/components/settings/LocationInput.tsx:14]: switched to an https api

**Comment by jfschwarz** [comment]
> aspect lines that are within ~5 points of being perfectly horizontal or vertical mess up the gradients applied to the aspect lines -__-

This one is a bit tricky but I found a hack to workaround the problem of degenerate bounding boxes by using paths instead of lines and adding an invisible 25x25 square to the bounding boxes of each line. 🤷  Wdyt, @samepant?

**Review by samepant — APPROVED** [review]
Looks good!!

**Pull request merged**